generators/upgrade/index.js

```diff
@@ -79,7 +79,8 @@ module.exports = UpgradeGenerator.extend({
         this.log(`Regenerating application with JHipster ${version}...`);
         let generatorCommand = 'yo jhipster';
         if (semver.gte(version, FIRST_CLI_SUPPORTED_VERSION)) {
-            generatorCommand = this.clientPackageManager === 'yarn' ? '$(yarn bin)/jhipster' : '$(npm bin)/jhipster';
+            const generatorDir = this.clientPackageManager === 'yarn' ? shelljs.exec('yarn bin') : shelljs.exec('npm bin');
```

Comment:
Also according to the [doc](https://github.com/shelljs/shelljs#execcommand--options--callback) sync exec should use stdout, though it seems to work without it.

shelljs.exec('yarn bin', {silent: true}).stdout;